File: examples/first-governed-action.py
import os
import asyncio
from dashclaw import DashClaw

async def run():
    # 1. Initialize DashClaw
    # In "demo" mode, it sends telemetry to the public DashClaw demo.
    claw = DashClaw(
        api_key=os.getenv("DASHCLAW_API_KEY", "demo"),
        base_url=os.getenv("DASHCLAW_BASE_URL", "http://localhost:3000")
    )

    print("🚀 Agent attempting high-risk action...")

    # 2. Intercept before you act
    # This sends the intent to DashClaw for policy evaluation.
    result = await claw.guard(
        action_type="deploy",
        risk_score=92,
        declared_goal="Deploy build v2.1.0 to production environment",
        reasoning="The build has passed all CI checks and is ready for release."
    )

    decision = result.get("decision", "unknown")
    action_id = result.get("actionId")

    print(f"⚖️ DashClaw decision: {decision.upper()}")
    
    if action_id:
        print(f"🔗 View decision replay: http://localhost:3000/decisions/{action_id}")

    # 3. Follow the decision
    if decision == "allowed":
        print("✅ Action permitted. Proceeding with deployment.")
    elif decision == "require_approval":
        print("⏳ Action paused. Awaiting human operator approval in Mission Control.")
    else:
        print("🛑 Action BLOCKED by governance policy.")

if __name__ == "__main__":
    try:
        asyncio.run(run())
    except Exception as e:
        print(f"❌ Error running example: {e}")
        print("\nTip: Make sure DashClaw is running locally at http://localhost:3000")
        print("Or run with: DASHCLAW_BASE_URL=https://www.dashclaw.io python first-governed-action.py")

File: examples/first-governed-action.js
import { DashClaw } from "dashclaw";

// 1. Initialize DashClaw
// If running locally, ensures it points to your local instance.
// In "demo" mode, it sends telemetry to the public DashClaw demo.
const claw = new DashClaw({
  apiKey: process.env.DASHCLAW_API_KEY || "demo",
  baseUrl: process.env.DASHCLAW_BASE_URL || "https://your-dashclaw.vercel.app",
  agentId: process.env.DASHCLAW_AGENT_ID || "first-action-agent"
});

async function run() {
  console.log("🚀 Agent attempting high-risk action...");

  // 2. Intercept before you act
  // This sends the intent to DashClaw for policy evaluation.
  const { decision, action_id } = await claw.guard({
    action_type: "deploy",
    risk_score: 92,
    declared_goal: "Deploy build v2.1.0 to production environment",
    reasoning: "The build has passed all CI checks and is ready for release."
  });

  console.log(`⚖️ DashClaw decision: ${decision.toUpperCase()}`);
  
  if (action_id) {
    console.log(`🔗 View decision replay: http://localhost:3000/decisions/${action_id}`);
  }

  // 3. Follow the decision
  if (decision === "allow") {
    console.log("✅ Action permitted. Proceeding with deployment.");
  } else if (decision === "require_approval") {
    console.log("⏳ Action paused. Awaiting human operator approval in Mission Control.");
  } else {
    console.log("🛑 Action BLOCKED by governance policy.");
  }
}

run().catch(err => {
  console.error("❌ Error running example:", err.message);
  console.log("\nTip: Make sure DashClaw is running locally at http://localhost:3000");
  console.log("Or run with: DASHCLAW_BASE_URL=https://your-dashclaw.vercel.app node first-governed-action.js");
});
